plugins/anomaly-detection/anomaly_detection.py

```diff
@@ -417,22 +661,52 @@ def anomaly_detection(algo_type, db_jdbc_address, start_time, end_time,
             "query", sql_query).load()
     )
 
-    prepared_DF = initDF.groupby(DF_GROUP_COLUMNS).agg(
-        f.collect_list("flowEndSeconds").alias("flowEndSeconds"),
-        f.stddev_samp("max(throughput)").alias("throughputStandardDeviation"),
-        f.collect_list(f.struct(["Diff_Secs", "max(throughput)"])).alias(
-            "Diff_Secs, Throughput"))
+    if agg_flow:
+        if agg_flow == "pod":
+            if pod_label:
+                df_agg_grp_columns = DF_AGG_GRP_COLUMNS_POD
+            elif pod_name:
+                df_agg_grp_columns = DF_AGG_GRP_COLUMNS_PODNAME
```

Comment:
If I understand correctly, in the case of pod_label and pod_name are both empty, we will need to set df_agg_grp_columns to DF_AGG_GRP_COLUMNS_POD.
If so I would suggest:
```         
            if pod_name:
                df_agg_grp_columns = DF_AGG_GRP_COLUMNS_PODNAME
            else:
                df_agg_grp_columns = DF_AGG_GRP_COLUMNS_POD
```